saleor/graphql/app/types.py

```diff
@@ -132,65 +149,83 @@ class Meta:
     def resolve_target(root, _info: ResolveInfo):
         return root.get("target") or AppExtensionTarget.POPUP
 
+    @staticmethod
+    def resolve_target_name(root, _info: ResolveInfo):
+        return root.get("target") or AppExtensionTarget.POPUP
+
+    @staticmethod
+    def resolve_mount_name(root, _info: ResolveInfo):
+        return root.get("mount")
+
     @staticmethod
     def resolve_url(root, _info: ResolveInfo):
         """Return an extension URL."""
         return resolve_app_extension_url(root)
 
 
+# Deprecated - typed options will be changed to JSON object
 class HttpMethod(BaseEnum):
     POST = AppExtensionHttpMethod.POST
     GET = AppExtensionHttpMethod.GET
 
 
+# Deprecated - typed options will be changed to JSON object
 class NewTabTargetOptions(BaseObjectType):
     method = graphene.Field(
         HttpMethod,
         required=True,
         description="HTTP method for New Tab target (GET or POST)",
+        deprecation_reason="Options will be changed to JSON object and contract will be held by Dashboard.",
```

Comment:
Nitpick: As Maciej noticed, from the end-user perspective, it should say that the dashboard holds the contract, not "will hold" (it already does, and the user does not have to think when it will hold it in the future). Also, let's use one way of referring to the dashboard ("Dashboard" vs  "the dashboard")